examples/decoding/cohy_decomposition.py

```diff
@@ -0,0 +1,532 @@
+# offered in :func:`~mne_connectivity.spectral_connectivity_epochs` and
+# :func:`~mne_connectivity.spectral_connectivity_time`, the filters are fit for each
+# frequency separately, and the filters are only applied to the same data they are fit
+# on.
+#
+# Unfortunately, fitting filters for each frequency bin can be computationally
+# expensive, which may prohibit the use of these techniques, e.g. in real-time BCI
+# setups where the rapid analysis of data is paramount, or even in offline analyses
+# with huge datasets.
+#
+# These issues are addressed by the :class:`~mne_connectivity.decoding.CaCoh` and
+# :class:`~mne_connectivity.decoding.MIC` decomposition classes of the decoding module.
+# Here, the filters are fit for a given frequency band collectively (not each frequency
+# bin!) and are stored, allowing them to be applied to the same data they were fit on
+# (e.g. for offline analyses of huge datasets) or to new data (e.g. for online analyses
+# of streamed data).
+#
+# In this example, we show how the tools of the decoding module compare to the standard
+# ``spectral_connectivity_...()`` functions in terms of their run time, and their
+# ability to decompose data into connectivity components.
+
+########################################################################################
+# Case 1: Fitting to and transforming different data
+# --------------------------------------------------
+#
+# We start by simulating some connectivity between two groups of signals at 15-20 Hz as
+# 60 two-second-long epochs. To demonstrate the approach of fitting filters to one set
+# of data and applying to another set of data, we will treat the first 30 epochs as the
+# data on which we train the filters, and the last 30 epochs as the data we transform.
+# We will use the CaCoh method, since zero time-lag interactions are not present (See
+# :doc:`../compare_coherency_methods` for more information).
+
+# %%
+
+N_SEEDS = 10
+N_TARGETS = 15
+
+FMIN = 15
+FMAX = 20
+
+N_EPOCHS = 60
+
+epochs = make_signals_in_freq_bands(
+    n_seeds=N_SEEDS,
+    n_targets=N_TARGETS,
+    freq_band=(FMIN, FMAX),
+    n_epochs=N_EPOCHS,
+    n_times=200,
+    sfreq=100,
+    snr=0.2,
+    rng_seed=44,
+)
+
+indices = (np.arange(N_SEEDS), np.arange(N_TARGETS) + N_SEEDS)
+
+########################################################################################
+# First, we use the standard CaCoh approach in
+# :func:`~mne_connectivity.spectral_connectivity_epochs` to visualise the connectivity
+# in the first 30 epochs. We also plot bivariate coherence to demonstrate the
+# signal-to-noise enhancements this multivariate approach offers. As expected, we see a
+# peak in connectivity at 15-20 Hz decomposed by the spatial filters.
+
+# %%
+
+# Connectivity profile of first 30 epochs (filters fit to these epochs)
+con_cacoh_first = spectral_connectivity_epochs(
+    epochs.get_data(item=np.arange(N_EPOCHS // 2)),
```

Comment:
as an ignorant outsider: why does a function that is called `spectral_connectivity_epochs` take a numpy array as first argument and not an `mne.Epochs` object?

Reply:
`spectral_connectivity_epochs` and `spectral_connectivity_time` both require epoched data, either as an `mne.Epochs` object or a 3D NumPy array.

Here I am passing in the array because I also want to give only the first 30 epochs, and using this item indexing with `.get_data()` is quicker than splitting the data into two separate `Epochs` objects for the first and last 30 epochs.

Reply:
You can index `Epoch` objects, so to get the first 30 epochs, you simply do `epochs[:30]`, which is super quick.

Reply:
Quite right, I always forget these alternative indexing options. Have changed.